docs/reference.cue

```diff
@@ -1 +1,36 @@
 package metadata
+
+#Fields: [Name=string]: #Fields | _
+
+remap: {
+  errors: [Name=string]: {
+    description: string
+    name: Name
+  }
+
+  functions: [Name=string]: {
+    arguments: [
```

Comment:
Not sure how to represent this yet, but arguments can have additional validation other than its type signature.

For example, we can force an integer to be within a certain range, or a string to be a fixed set of values (i.e. enumerated type).

Perhaps we should add a free-form text field to describe conditionals like these? Although for enums, I do think it makes sense to explicitly mention those, as we do with regular enum-like component options as well.